### DISCUSSION TITLE
setting logging level

### DISCUSSION THREAD

**Discussion by gdevelekos** [jaegertracing/jaeger#3658] [Q&A]
I'm trying to reduce logging to "error" (or perhaps "warning") level for the Jaeger Collector pod.

In the values file, in all sections (just to eliminate unlucky misses) wherever I saw: 

`cmdlineParams: {}`

...I have replaced it with:

```
cmdlineParams: 
    log-level: "error"
```

...and yet when the deployment succeeds, the container still logs at the "info" level.

```
2022/05/06 14:11:41 maxprocs: Updating GOMAXPROCS=1: using minimum allowed GOMAXPROCS
{"level":"info","ts":1651846301.869269,"caller":"zapgrpc/zapgrpc.go:129","msg":"Deprecation warning: 299 Elasticsearch-7.16.3-4e6e4eab2297e949ec994e688dad46290d018022 \"legacy template [jaeger-span] has index patterns [*jaeger-span-*] matching patterns from existing composable templates [.deprecation-indexing-template,.ml-anomalies-,.ml-state,.ml-stats,.slm-history,.watch-history-13,ilm-history,logs,metrics,synthetics] with patterns (.deprecation-indexing-template => [.logs-deprecation.*],.ml-anomalies- => [.ml-anomalies-*],.ml-state => [.ml-state*],.ml-stats => [.ml-stats-*],.slm-history => [.slm-history-5*],.watch-history-13 => [.watcher-history-13*],ilm-history => [ilm-history-5*],logs => [logs-*-*],metrics => [metrics-*-*],synthetics => [synthetics-*-*]); this template [jaeger-span] may be ignored in favor of a composable template at index creation time\""}
{"level":"info","ts":1651846301.8693128,"caller":"zapgrpc/zapgrpc.go:129","msg":"Deprecation warning: 299 Elasticsearch-7.16.3-4e6e4eab2297e949ec994e688dad46290d018022 \"Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.16/security-minimal-setup.html to enable security.\""}
{"level":"info","ts":1651846301.8693204,"caller":"zapgrpc/zapgrpc.go:129","msg":"Deprecation warning: 299 Elasticsearch-7.16.3-4e6e4eab2297e949ec994e688dad46290d018022 \"Legacy index templates are deprecated in favor of composable templates.\""}
```

- A Jaeger newbie :-)

**Comment by albertteoh** [comment]
These are logged by an independent logger for the ES client, whose level can be configured via the `es.log-level` parameter (i.e. `es.log-level: "error"`. This is the default, so no need for explicit config). However, it turns out this isn't so helpful at the moment as it seems there's a bug in its configuration.

This PR aims to address that bug: https://github.com/jaegertracing/jaeger/pull/3664.

**Reply by gdevelekos** [reply]
Thanks for answering. Are you saying that, until that bug is fixed, there is no way to drop the logging level from "info" to "error" or "warning" for all logging in the pod?  Will my config do the trick once the bug is fixed? Or must I change log-level to es.log-level in some (or all) places?

**Reply by albertteoh** [reply]
> Will my config do the trick once the bug is fixed? 

Yes, your existing config of `log-level: "error"` will do the trick once you upgrade to the next release (v1.34).

> there is no way to drop the logging level from "info" to "error" or "warning" for all logging in the pod? 

Until then, you could try:
```
log-level: "error"
es.log-level: "info"
```

ES info logs are a little less "wordy"; you won't get the verbose deprecation notices.

**Reply by gdevelekos** [reply]
Thanks again! Please give me an estimate on the release date of v1.34.

**Reply by albertteoh** [reply]
The release should be cut by the end of this week.
